edx_proctoring/utils.py

```diff
@@ -395,3 +403,58 @@ def categorize_inaccessible_exams_by_date(onboarding_exams, details):
                 non_date_inaccessible_exams.append(onboarding_exam)
 
     return non_date_inaccessible_exams, future_exams, past_due_exams
+
+
+def encrypt_and_encode(data, key):
+    """ Encrypts and encodes data using a key """
+    return base64.urlsafe_b64encode(aes_encrypt(data, key))
+
+
+def decode_and_decrypt(encoded_data, key):
+    """ Decrypts and decodes data using a key """
+    return aes_decrypt(base64.urlsafe_b64decode(encoded_data), key)
+
+
+def aes_encrypt(data, key):
+    """
+    Return a version of the data that has been encrypted
+    """
+    cipher = Cipher(AES(key), CBC(generate_aes_iv(key)), backend=default_backend())
+    padded_data = pad(data)
+    encryptor = cipher.encryptor()
+    return encryptor.update(padded_data) + encryptor.finalize()
```

Comment:
It's absolutely fascinating to me that the api being used here uses addition in this way. I haven't dug far enough into the documentation to understand why it would be, but I trust that this is what we were doing before. Just... so weird.